rfc/rfc011-verifiable-credential.md

```diff
@@ -0,0 +1,215 @@
+
+## 1. Introduction
+
+It's to be expected that multiple sources must be supported that can claim certain information about a subject. 
+These claims have to be structured in such a way, that they are verifiable and searchable. 
+Examples of claims are a company name, and it's chamber of commerce number. Users will search on the name of an organization to find out which services are supported.
+This will directly influence the interaction the user can have with the system of that organization. 
+Making sure this information is correct and that it can be trusted is therefore extremely important.
+
+## 2. Terminology
+
+* **DID**: [Decentralized Identifiers](https://www.w3.org/TR/did-core/).
+* **holder**: The party that receives a VC from an issuer.  
+* **issuer**: The party that issues and signs a VC.  
+* **proof**: Proof asserting that a VC is valid using cryptography.
+* **VC**: Verifiable Credential according to the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+* **VP**: Verifiable Presentation according to the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+
+## 3. W3C Verifiable Credential
+
+To support a variety of claims, the [W3C Verifiable Credential specification](https://www.w3.org/TR/vc-data-model/) is used. 
+Every Nuts specific VC and proof type must follow the W3C specification. 
+If the W3C specification offers options, then the specific VP must specify which option is to be used.
+All VCs MUST use DIDs as specified in [RFC004](rfc004-distributed-document-format.md) for the issuer. 
+
+### 3.1 Supported proofs
+
+The following proof types must be supported:
+
+#### 3.1.1 JsonWebSignature2020
+
+resources:
+
+* https://w3c-ccg.github.io/ld-proofs/ 
+* https://tools.ietf.org/html/rfc7797
+* https://w3c-ccg.github.io/lds-jws2020
+
+The algorithm used to create the signature is as follows:
+
+- take the input Verifiable Credential and remove the `proof` field. Store as `raw_vc`.
+- compute the sha256 of the `raw_vc`, store as `hash_vc = sha256(raw_vc)`.  
+- compose a `raw_proof` json object without the `jws` field:
+
+```json
+{
+  "type": "JsonWebSignature2020",
+  "proofPurpose": "assertionMethod",
+  "verificationMethod": "did:nuts:EgFjg8zqN6eN3oiKtSvmUucao4VF18m2Q9fftAeANTBd#twlH6rB8ArZrknmBRWLXhao3FutZtvOm0hnNhcruenI",
+  "created": "2021-03-09T11:44:56.382202+01:00"
+}
+```
+where the `verificationMethod` must be a valid assertionMethod ID from the DID Document. `created` MUST be a RFC3339 compliant time string.
+`type` and `proofPurpose` MUST be entered as above.
+
+- compute the sha256 of the `raw_proof`, store as `hash_proof = sha256(raw_proof)`.
+- concatenate  `hash_proof` with `raw_proof` and store as `payload`.
+- construct the JWS `header` as `{"alg":"ES256","b64":false,"crit":["b64"]}` where the `alg` SHOULD be replaced with a valid algorithm.
```

Comment:
should because ES256 could already be the correct algorithm? Then I'd say `alg` MUST contain the correct algorithm.